### ISSUE TITLE
Can’t access guacamole system

### ISSUE THREAD

**Issue by Nbr1Sniper** [bleakvin/hass-guacamole#4]
Multiple people have reported on your home assistant page and error:

11:53:29.854 [http-nio-8080-exec-3] INFO  o.a.g.r.auth.AuthenticationService - User "guacadmin" successfully authenticated from [127.0.0.1, 172.30.32.1, 172.30.32.2].
11:53:30.296 [http-nio-8080-exec-1] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging for details.
11:53:30.302 [http-nio-8080-exec-10] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging for details.

**Comment by egandro** [comment]
Hello @Nbr1Sniper I can confirm an issue:

I enabled the access to 8080 to make this more clear.

`http://homeassistant.local:8080/api/tokens`

And I got the following docker logs

`18:19:07.474 [http-nio-8080-exec-10] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging
`

Diging deeper...

**Comment by Nbr1Sniper** [comment]
@egandro thank you!!!!

**Comment by egandro** [comment]
Debugging ....

```
URL: https://foobar/api/hassio_ingress/XXXXXXXXXXX/api/session/data/postgresql/connectionGroups/ROOT/tree?token=whatever

{"message":"Unexpected internal error","translatableMessage":{"key":"APP.TEXT_UNTRANSLATED","variables":{"MESSAGE":"Unexpected internal error"}},"statusCode":null,"expected":null,"type":"INTERNAL_ERROR"}
```

The logfile has this entries:

```
22:21:23.156 [http-nio-8080-exec-10] INFO  o.a.g.r.auth.AuthenticationService - User "guacadmin" successfully authenticated from [46.114.92.29, 192.168.10.1, 172.30.32.1, 172.30.32.2].
22:21:23.832 [http-nio-8080-exec-6] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging for details.
```
So according to the documentation I should change the level "info" to "debug". Here the documentation: https://guacamole.apache.org/doc/0.9.6/gug/configuring-guacamole.html

Here what is inside the container:

```
\u279c  ~ docker exec -it addon_54305586_haguacamole /bin/bash
root@54305586-haguacamole:/config# tail -10 /usr/local/tomcat/webapps/ROOT/WEB-INF/classes/logback.xml
            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
        </encoder>
    </appender>

    <!-- Log at INFO level -->
    <root level="info">
        <appender-ref ref="GUAC-DEFAULT" />
    </root>

</configuration>
```

Unfortunately I am to stupid to do this. After restarting the container the changes are gone and I have no Idea how to debug docker containers in the realm of home assistant.

Giving up at this point until someone finds an idea.

Using port :8080 works.

**Comment by Nbr1Sniper** [comment]
I found if I turn on 8080 as you mentioned I can access from a browser but I can’t access from home assistant. Thanks for the help.

**Comment by Devvz** [comment]
Not sure why this was closed, as I'm also having this issue along with a group of people in this [thread](https://community.home-assistant.io/t/new-addon-access-machines-in-your-homenet-with-apache-guacamole-ingress-support/209829/14). Seems that @bleakvin is aware, but might not be able to fix any time soon.

I tried digging around, but don't really have much experience with docker to enable the debug logs to see what the error **actually** is.

Hopefully someone more familiar with docker could take a look...

**Comment by bleakvin** [comment]
Yeah the issue definetly persists. I tried to update the base image and some components in the container but ran into a whole slew of other issues. Unfortunately I'm not sure how to progress from here.

**Comment by Devvz** [comment]
> 
> 
> Yeah the issue definetly persists. I tried to update the base image and some components in the container but ran into a whole slew of other issues. Unfortunately I'm not sure how to progress from here.

Are you able to change the logs to debug? What is the actual error that is reported?

**Comment by bleakvin** [comment]
Another issue that just occured to me is that Dockerhub changed their policy for free accounts, so I won't even be able to build the Docker Image for the different platforms. I guess that could be done with Github actions instead but I have no clue on 
how to do that. 😅

**Comment by egandro** [comment]
@bleakvin are you on the Discord servers of HA? What timezone do you live? (I am from EU). We might work on this together.

Guacamole is a beast :( From what I have read it might also has to do with timezones (!) in the Database.

**Comment by bleakvin** [comment]
I managed to enable the debugging but cant really read something usefull out of it🤷‍♀️

```
12:33:36.655 [http-nio-8080-exec-4] INFO  o.a.g.r.auth.AuthenticationService - User "guacadmin" successfully authenticated from [##########, 172.30.32.1, 172.30.32.2].
12:33:36.673 [http-nio-8080-exec-4] DEBUG o.a.g.a.f.FileAuthenticationProvider - User mapping file "/data/guacamole/user-mapping.xml" does not exist and will not be read.
12:33:36.688 [http-nio-8080-exec-4] DEBUG o.a.g.r.auth.AuthenticationService - Login was successful for user "guacadmin".
12:33:36.750 [http-nio-8080-exec-3] DEBUG o.a.g.rest.RESTExceptionMapper - Client request rejected: No such token.
12:33:36.867 [http-nio-8080-exec-8] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging for details.
12:33:36.868 [http-nio-8080-exec-10] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging for details.
12:33:36.869 [http-nio-8080-exec-1] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging for details.
12:33:36.871 [http-nio-8080-exec-9] ERROR o.a.g.rest.RESTExceptionMapper - An internal error occurred, but did not contain an error message. Enable debug-level logging for details.
12:33:36.874 [http-nio-8080-exec-8] DEBUG o.a.g.rest.RESTExceptionMapper - Unexpected error in REST endpoint.
javax.ws.rs.WebApplicationException: null
	at com.sun.jersey.server.impl.uri.rules.TerminatingRule.accept(TerminatingRule.java:66)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1511)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1442)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1391)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1381)
	at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:416)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:538)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:716)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
	at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:263)
	at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:178)
	at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:91)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:62)
	at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:118)
	at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:113)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:200)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:490)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:836)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1839)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)
12:33:36.875 [http-nio-8080-exec-1] DEBUG o.a.g.rest.RESTExceptionMapper - Unexpected error in REST endpoint.
javax.ws.rs.WebApplicationException: null
	at com.sun.jersey.server.impl.uri.rules.TerminatingRule.accept(TerminatingRule.java:66)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1511)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1442)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1391)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1381)
	at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:416)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:538)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:716)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
	at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:263)
	at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:178)
	at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:91)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:62)
	at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:118)
	at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:113)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:200)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:490)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:836)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1839)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)
12:33:36.893 [http-nio-8080-exec-10] DEBUG o.a.g.rest.RESTExceptionMapper - Unexpected error in REST endpoint.
javax.ws.rs.WebApplicationException: null
	at com.sun.jersey.server.impl.uri.rules.TerminatingRule.accept(TerminatingRule.java:66)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1511)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1442)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1391)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1381)
	at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:416)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:538)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:716)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
	at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:263)
	at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:178)
	at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:91)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:62)
	at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:118)
	at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:113)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:200)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:490)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:836)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1839)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)
12:33:36.893 [http-nio-8080-exec-9] DEBUG o.a.g.rest.RESTExceptionMapper - Unexpected error in REST endpoint.
javax.ws.rs.WebApplicationException: null
	at com.sun.jersey.server.impl.uri.rules.TerminatingRule.accept(TerminatingRule.java:66)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1511)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1442)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1391)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1381)
	at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:416)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:538)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:716)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
	at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:263)
	at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:178)
	at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:91)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:62)
	at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:118)
	at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:113)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:200)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:490)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:836)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1839)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)

**Comment by bleakvin** [comment]
> @bleakvin are you on the Discord servers of HA? What timezone do you live? (I am from EU). We might work on this together.

Im not active on the Discord Server, just reading occasionally. But I am from the eu as well (Germany to be precise)

> Guacamole is a beast :( From what I have read it might also has to do with timezones (!) in the Database.

Yeah and I dont really know anything about the inner workings of it to be honest. I just took an existing standalone container for it and added some modifications for it to work with Homeassistant.

**Comment by Devvz** [comment]
From the research that I've done, it seems that there is a null value somewhere that shouldn't be. I got that from the following [thread](https://www.mail-archive.com/user@guacamole.apache.org/msg03904.html).

Looking at the `config.json` there's the following:
```json    
"ports": {
    "8080/tcp": null
},
```
and then below it,
```json
"ports_description": {
      "8080/tcp": "Web interface (Not required for Home Assistant Ingress)"
},
```

Maybe it doesn't like the `12:33:36.893 [http-nio-8080-exec-9] DEBUG o.a.g.rest.RESTExceptionMapper`? Just throwing out ideas, honestly... I have no idea what homeassistant needs for ingress to work. I suppose I'll look more into that.

**Comment by egandro** [comment]
I guess we have some issues with some Java security BS :(

**Comment by egandro** [comment]
@bleakvin you want to try this?

https://developers.home-assistant.io/docs/add-ons/presentation/#ingress

Ingeress might be needs some extra hugs for the websocket of guacamole

**Comment by egandro** [comment]
Yesss ... it uses a Websocket: 

`ws://192.168.100.200:8080/websocket-tunnel`

Trying to fix this.

**Comment by iridris** [comment]
Any progress on this issue?

**Comment by egandro** [comment]
> Any progress on this issue?

Well according to the home assistant manual, websockets "should be" tunneled. But they are not (I might be wrong and the issue is by a broken route).

The problem is also a double nginx and bloated Java.

We have:

- The world. 
- Maybe some reverse proxy + SSL certificate
- Maybe some router at home connecting to the reverse proxy
- Home Assistant
- Docker/Ingress Nginx
- HAAS Guacamole Container
- Nginx
- bloated Java (with all sort's of it's own things) that starts a Websocket.

We need to pierce "the world".

I also debugged it down, where the API thing (that gets a 503) expects a different route for a simple GET /.../api/something request.

This route is wrong. The docker container name is added. There is a way to change this in the config.json. However I found out that same route neets to be inside the docker container somewhere in some (bloated) Java .env/.pref/.config file.

I edited this - however - the Docker container needs recompile or a wrapper and it doesn't persist my changes (and I am to stupid at the moment to use -v... from the config.json) Here I gave up.

**Comment by iridris** [comment]
@egandro, there are some commits in the main Guacamole repository that deal with URL parameters being appended improperly. Sounds like that may be related to the 503 you were seeing? It doesn't look like those commits have made it into a release of Guacamole yet, so would likely need to build guac from the source rather than use a release build. But I have no clue how to do that or test it.

Here's one of the commits, with 3 related others on the same day: https://github.com/apache/guacamole-server/commit/7f5539930456d4a02a5d9bf479981b2282de4ad6

**Comment by cnf** [comment]
Hi everyone, is there any update on this?

Thanks!

**Comment by egandro** [comment]
@iridris  and @cnf thx a lot.

I am still throwing sticks on a futuristic space shuttle here :( No idea what and how to debug.

https://guacamole.apache.org/releases/ ... 1.3.0 is still the most recent version of guacamole and the version that the docker image uses.

**Comment by bleakvin** [comment]
Hey everyone,

I tried again to fix this issue with version 1.4.0.
Unfortunately I cant seem to figure it out and I dont have the time to dig deeper into it.
I will archive this repository.
